src/main/cpp/syslogappender.cpp

```diff
@@ -322,33 +323,37 @@ void SyslogAppender::append(const spi::LoggingEventPtr& event, Pool& p)
 
 	if ( msg.size() > maxMessageLength )
 	{
+		int numberOfFullChunks = msg.size() / maxMessageLength;
 		LogString::iterator start = msg.begin();
+		LogString::iterator end = msg.begin() + maxMessageLength;
```

Comment:
> // Note: MSVC/Windows does not like it when you seek past the end of the string

Doesn't that comment apply for this line as well? Shouldn't there be a check for number of chunks aka the length of the string being greater than `maxMessageLength`?

    LogString::iterator end = numberOfFullChunks > 0|1 ? (msg.begin() + maxMessageLength) : msg.end();

Not sure about the case when message length and `maxMessageLength` are equal.

Reply:
Unless I have something wrong with my math, that line of code should only be relevant if the message is larger than the maximum message size(in which case it needs to split it up into several chunks).

Reply:
I didn't recognize the `if` before. :-)